openghg/standardise/boundary_conditions/_cams.py

```diff
@@ -0,0 +1,374 @@
+    Returns:
+        dict with 2 keys : "time" and "closed; whose values are the one passed to xr.Dataset().resample
+    """
+    org_freq_str = xr.infer_freq(xr_time)
+    org_freq = pd.to_timedelta(org_freq_str if org_freq_str[0].isdecimal() else "1" + org_freq_str)
+    thres_dict = {
+        "3h": pd.Timedelta(3, "h"),
+        "daily": pd.Timedelta(1, "D"),
+        "monthly": pd.Timedelta(31, "D"),
+        "yearly": pd.Timedelta(365, "D"),
+    }
+
+    if org_freq > thres_dict.get(period, period):
+        raise ValueError("Original time resolution is coarser than targeted one.")
+
+    alias_period = {"daily": "D", "monthly": "MS", "yearly": "YS"}
+
+    closed: Literal["left", "right"] = "right" if species == "n2o" else "left"
+    return {"time": alias_period.get(period, period), "closed": closed}
+
+
+def calc_altitude_from_pressure(ds: xr.Dataset) -> xr.Dataset:
+    """
+    Calculate altitude of dataset levels. Specific to CAMS N2O files.
+    Args:
+        ds: dataset (CAMS N2O files), contains ap, bp, and Psurf variables
+    Returns:
+        ds: dataset whith new variable altitude (in meter)
+    """
+    ds["pressure"] = ds.ap + ds.bp * ds.Psurf
+    scale_height = 7.64e3  # in metres
+    ds["altitude"] = -scale_height * np.log(ds.pressure / ds.Psurf)
+    del ds["pressure"]
+    return ds
+
+
+def get_gridsize(ds: xr.Dataset) -> str:
+    """
+    Calculate mean grid cell size of dataset
+    Args:
+        ds: dataset containing lon and lat coordinates
+    Returns
+        string of the form "{lat_grid} x {lon_grid}"
+    """
+    lat_grid = (ds["lat"].values[1:] - ds["lat"].values[:-1]).mean()
+    lon_grid = (ds["lon"].values[1:] - ds["lon"].values[:-1]).mean()
+    return f"{lat_grid} x {lon_grid}"
+
+
+def _check_and_set_params(
+    filepath: str | Path | list[str] | list[Path],
+    cams_version: str | None = None,
+    species: str | None = None,
+    input_observations: str | None = None,
+) -> tuple[list[Path], str, str, str]:
+    """
+    Extract fron the file names informations on species, cams version, input observations used in the files.
+    Check that they correspond to parameters cams_version, species and input_observations (if given) and format them to be saved as attributes and metadata.
+    Args:
+        filepath: list of filepath that will be standardised
+        cams_version: version of cams product that is used
+        secies: species of interest
+        input_observations: type of input observations (e.g. "surface_satellite_dm") that is used
+    Returns:
+        tuple containing 1. a string aggregating the cams versions used, 2. the species and 3. a string aggregating the input observations used.
+    """
+    new_filepath, cams_version_check, species_check, input_observations_check = [], [], [], []
+
+    if not isinstance(filepath, list):
+        filepath = [
+            filepath,
+        ]
+
+    for file in filepath:
```

Comment:
try the suggestions above, but if that doesn't work, do this:
```suggestion
    for file in filepath:  # type: ignore
```